demisto_sdk/commands/common/update_id_set.py

```diff
@@ -1445,12 +1535,25 @@ def find_duplicates(id_set, print_logs):
     return lists_to_return
 
 
-def has_duplicate(id_set, id_to_check, object_type=None, print_logs=True):
-    duplicates = [duplicate for duplicate in id_set if duplicate.get(id_to_check)]
+def has_duplicate(id_set_subset_list, id_to_check, object_type=None, print_logs=True, external_object=None):
+    """
+    Finds if id_set_subset_list contains a duplicate items with the same id_to_check.
+
+    Pass `external_object` to check if it exists in `id_set_subset_list`.
+    Otherwise the function will check if `id_set_subset_list` contains 2 or more items with the id of `id_to_check`
+
+    """
+    duplicates = [duplicate for duplicate in id_set_subset_list if duplicate.get(id_to_check)]
+
+    if external_object and len(duplicates) == 0:
+        return False
 
-    if len(duplicates) < 2:
+    if not external_object and len(duplicates) == 1:
         return False
 
+    if external_object:
+        duplicates.append(external_object)
```

Comment:
you can move this right after the list-comprehension and keep the original condition (instead of the above two)